File: algorithmic_toolbox/week1_programming_challenges/2_maximum_pairwise_product/python/test_max_pairwise_product.py
from max_pairwise_product import max_pairwise_product


def test_max_pairwise_product() -> None:
    assert max_pairwise_product([5, 6, 7, 4, 2]) == 6 * 7, "should calculate max pairwise product"
    assert (
        max_pairwise_product([90_000, 100_000]) == 9_000_000_000
    ), "should calculate max pairwise product for large numbers"

File: algorithmic_toolbox/week1_programming_challenges/2_maximum_pairwise_product/java/src/TestMaxPairwiseProduct.java
public class TestMaxPairwiseProduct {
  private static MaxPairwiseProduct maxPairwiseProduct = new MaxPairwiseProduct();

  public static void main(String[] args) {
    long[] numbers1 = { 5, 6, 7, 4, 2 };
    assert maxPairwiseProduct.getMaxPairwiseProduct(numbers1) == 6 * 7;

    long[] numbers2 = { 100000, 90000 };
    assert maxPairwiseProduct.getMaxPairwiseProduct(numbers2) == 100000 * 90000;
    System.out.println("All tests passed!");
  }
}
